service/src/main/java/bio/terra/axonserver/utils/CloudStorageUtils.java

```diff
@@ -77,33 +85,47 @@ public static byte[] getBucketObject(
             .getService()
             .reader(blobId)) {
 
-      long startByteIdx =
-          Optional.ofNullable(byteRange).isPresent() ? byteRange.getRangeStart(Long.MAX_VALUE) : 0;
-      long endByteIdx =
-          Optional.ofNullable(byteRange).isPresent()
-              ? byteRange.getRangeEnd(Long.MAX_VALUE)
-              : Long.MAX_VALUE - 1;
-      long bytesToRead = endByteIdx - startByteIdx;
-
-      BoundedByteArrayOutputStream outputStream = new BoundedByteArrayOutputStream(MAX_OBJECT_SIZE);
-      ByteBuffer buffer = ByteBuffer.allocate((int) Math.min(MAX_BUFFER_SIZE, bytesToRead));
-
-      long totalBytesRead = 0;
-      int bytesRead;
-      reader.seek(startByteIdx);
-      while (totalBytesRead < bytesToRead && reader.read(buffer) > 0) {
-        bytesRead = buffer.position();
-
-        // write bytes in buffer to output stream
-        buffer.flip();
-        outputStream.write(buffer.array(), 0, buffer.limit());
-
-        // clear buffer and increment bytesRead
-        buffer.clear();
-        totalBytesRead += bytesRead;
+      // Create a temporary file to store the object contents and set its permissions to 700
+      Set<PosixFilePermission> permissions = PosixFilePermissions.fromString("rwx------");
+      File outputFile =
+          File.createTempFile(
+              FilenameUtils.getBaseName(objectName) + "-",
+              "." + FilenameUtils.getExtension(objectName),
+              new File(CreateDownloadsFolder.DOWNLOADS_DIR.toString()));
+      outputFile.deleteOnExit();
```

Comment:
looking at the doc for this, it will `Request that the file or directory denoted by this abstract pathname be deleted when the virtual machine terminates`.
I might be misunderstanding, but I think this means the files wouldn't be cleaned up until the Axon instance goes down, which means we would accumulate (potentially large) temporary files. Can you validate if this happens? if so, this might need a different approach

Reply:
Good catch :(. Looks like the behavior is as you described.

I just pushed a change to use a custom spring resource `CleanupInputStreamResource` that properly deletes the file after it is read. Tested and the files no longer live past the response.